pkg/datasource/endian_be.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build armbe || arm64be || mips || mips64 || mips64p32 || ppc64 || s390 || s390x || sparc || sparc64
+
+package datasource
+
+// NativeEndian is set to either binary.BigEndian or binary.LittleEndian,
+// depending on the host's endianness. Borrowed from cilium/ebpf.
+var NativeEndian = binary.BigEndian
```

Comment:
Since we use go 1.21 (see go.mod), we don't need `endian_be.go` and `endian_le.go` anymore. We can use `binary.NativeEndian`:
https://pkg.go.dev/encoding/binary#pkg-variables

See [go 1.21 release notes](https://tip.golang.org/doc/go1.21):
> [encoding/binary](https://tip.golang.org/pkg/encoding/binary/)
The new [NativeEndian](https://tip.golang.org/pkg/encoding/binary/#NativeEndian) variable may be used to convert between byte slices and integers using the current machine’s native endianness.

Reply:
TIL, nice! I've read a couple of rants why the Go authors decided not to introduce that a couple of years back, great that they changed their minds :)